--- argument_config_ori.py	2025-07-08 15:54:53.882968023 +0800
+++ argument_config.py	2025-07-08 15:54:18.228964976 +0800
@@ -1,3 +1,6 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
 # coding: utf-8
 
 """
@@ -16,6 +19,8 @@
     source: Annotated[str, tyro.conf.arg(aliases=["-s"])] = make_abs_path('../../assets/examples/source/s0.jpg')  # path to the source portrait (human/animal) or video (human)
     driving:  Annotated[str, tyro.conf.arg(aliases=["-d"])] = make_abs_path('../../assets/examples/driving/d0.mp4')  # path to driving video or template (.pkl format)
     output_dir: Annotated[str, tyro.conf.arg(aliases=["-o"])] = 'animations/'  # directory to save output video
+    output_name: Annotated[str, tyro.conf.arg(aliases=["-n"])] = 'video'  # name of the output video
+    template_dir: Annotated[str, tyro.conf.arg(aliases=["-t"])] = 'templates'  # name of the output video
 
     ########## inference arguments ##########
     flag_use_half_precision: bool = True  # whether to use half precision (FP16). If black boxes appear, it might be due to GPU incompatibility; set to False.
